pkg/admission/quota/clusterresourcequota/admission.go

```diff
@@ -93,12 +93,12 @@ func (q *clusterQuotaAdmission) Validate(ctx context.Context, a admission.Attrib
 	// to its name. Namespaces are cluster level object that shouldn't go into this plugin or it get blocked listing
 	// the namespace that is just being created.
 	obj := a.GetObject()
-	objMeta, ok := obj.(metav1.ObjectMetaAccessor)
-	if !ok {
-		klog.Warningf("ClusterQuotaAdmission received non-meta object %T", obj)
+	objMeta, err := metav1.Accessor(obj)
```

Comment:
the error was `... unstructured.Unstructured` which doesn't expose metadata so it won't help I think

Reply:
`Unstructured  exposes `GetNamespace()` method

Reply:
unless `Unstructured` implements `metav1.Object` interface - https://github.com/kubernetes/kubernetes/blob/e00bb664e1f58af5e3e9eb9c692bf9958a31cf7d/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go#L48

Reply:
@p0lyn0mial  It doesn't have to expose metadata if it implements Object (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go#L98) which Unstructured for CRDs does implement that and the accessor will work: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go#L86